src/GafferSceneUI/Inspector.cpp

```diff
@@ -269,6 +269,15 @@ void Inspector::inspectHistoryWalk( const GafferScene::SceneAlgo::History *histo
 		}
 	}
 
+	// If we're at the start of the history and haven't found a source, give an explanation.
+
+	if( !history->predecessors.size() && !result->m_source )
+	{
+		result->m_editFunction = "No editable source found in history.";
+
+		return;
+	}
+
```

Comment:
This seems to have broken the case below, where the original source node isn't editable, but there is an EditScope downstream of it. If you focus the EditScope, and choose to use the EditScope in the LightEditor, and then double-click a parameter, you get the "no editable source" warning rather than an editor popping up. Perhaps [this](https://github.com/GafferHQ/gaffer/blob/main/src/GafferSceneUI/Inspector.cpp#L161) is a better place to move the check, as an `else if` clause? I think that spot has the additional benefit of not getting confused if there is a branching history, and we bottom out in one branch before we would find a suitable edit in another branch.

It's a bit worrying that our existing unit tests weren't strong enough to pick up this regression, so would also be good to add a test that catches it.

<details>

```
import Gaffer
import GafferScene
import imath

Gaffer.Metadata.registerValue( parent, "serialiser:milestoneVersion", 1, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:majorVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:minorVersion", 0, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["SceneReader"] = GafferScene.SceneReader( "SceneReader" )
parent.addChild( __children["SceneReader"] )
__children["SceneReader"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["EditScope"] = Gaffer.EditScope( "EditScope" )
parent.addChild( __children["EditScope"] )
__children["EditScope"].addChild( Gaffer.BoxIn( "BoxIn" ) )
__children["EditScope"]["BoxIn"].setup( GafferScene.ScenePlug( "out", ) )
__children["EditScope"].addChild( GafferScene.ScenePlug( "in", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["EditScope"].addChild( Gaffer.BoxOut( "BoxOut" ) )
__children["EditScope"]["BoxOut"].setup( GafferScene.ScenePlug( "in", ) )
__children["EditScope"].addChild( GafferScene.ScenePlug( "out", direction = Gaffer.Plug.Direction.Out, flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["EditScope"].addChild( Gaffer.BoolPlug( "enabled", defaultValue = True, flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["EditScope"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["SceneReader"]["fileName"].setValue( '/home/john/dev/build/gaffer-1.0-Python3/python/GafferSceneTest/usdFiles/sphereLight.usda' )
__children["SceneReader"]["__uiPosition"].setValue( imath.V2f( 4.30000019, 3.1500001 ) )
__children["EditScope"]["BoxIn"]["__in"].setInput( __children["EditScope"]["in"] )
Gaffer.Metadata.registerValue( __children["EditScope"]["BoxIn"]["__in"], 'nodule:color', imath.Color3f( 0.240099996, 0.339399993, 0.485000014 ) )
Gaffer.Metadata.registerValue( __children["EditScope"]["BoxIn"]["__in"], 'layout:section', 'Settings' )
Gaffer.Metadata.registerValue( __children["EditScope"]["BoxIn"]["__in"], 'description', 'The output scene.' )
Gaffer.Metadata.registerValue( __children["EditScope"]["BoxIn"]["__in"], 'nodule:type', 'GafferUI::StandardNodule' )
__children["EditScope"]["in"].setInput( __children["SceneReader"]["out"] )
Gaffer.Metadata.registerValue( __children["EditScope"]["in"], 'nodule:color', imath.Color3f( 0.240099996, 0.339399993, 0.485000014 ) )
Gaffer.Metadata.registerValue( __children["EditScope"]["in"], 'description', 'The output scene.' )
Gaffer.Metadata.registerValue( __children["EditScope"]["in"], 'nodule:type', 'GafferUI::StandardNodule' )
__children["EditScope"]["BoxOut"]["in"].setInput( __children["EditScope"]["BoxIn"]["out"] )
Gaffer.Metadata.registerValue( __children["EditScope"]["BoxOut"]["__out"], 'nodule:color', imath.Color3f( 0.240099996, 0.339399993, 0.485000014 ) )
Gaffer.Metadata.registerValue( __children["EditScope"]["BoxOut"]["__out"], 'layout:section', 'Settings' )
Gaffer.Metadata.registerValue( __children["EditScope"]["BoxOut"]["__out"], 'description', 'The output scene.' )
Gaffer.Metadata.registerValue( __children["EditScope"]["BoxOut"]["__out"], 'nodule:type', 'GafferUI::StandardNodule' )
__children["EditScope"]["BoxOut"]["passThrough"].setInput( __children["EditScope"]["BoxIn"]["out"] )
__children["EditScope"]["BoxOut"]["enabled"].setInput( __children["EditScope"]["enabled"] )
__children["EditScope"]["out"].setInput( __children["EditScope"]["BoxOut"]["__out"] )
Gaffer.Metadata.registerValue( __children["EditScope"]["out"], 'nodule:color', imath.Color3f( 0.240099996, 0.339399993, 0.485000014 ) )
Gaffer.Metadata.registerValue( __children["EditScope"]["out"], 'description', 'The output scene.' )
Gaffer.Metadata.registerValue( __children["EditScope"]["out"], 'nodule:type', 'GafferUI::StandardNodule' )
__children["EditScope"]["__uiPosition"].setValue( imath.V2f( 5.79910135, -5.18203115 ) )


del __children
```

</details>

Reply:
I think I have it sorted out in 44028947c2622a6582562c822579c10eb94a8b00. I changed the condition somewhat, and also moved the existing test `if( !result->m_value && !result->editable() )` to be first. It seemed a little confusing that we could be modifying `result` in a conditional but then ignoring it altogether and bouncing out with `nullptr` anyways. Does that make sense?

Reply:
Yep, LGTM.